### ISSUE TITLE
Hyper Trained stats revert after re-loading .pk7

### ISSUE THREAD

**Issue by sora10pls** [kwsch/PKHeX#754]
Tested with latest commit.

Pokemon originated in Gen 6, was converted from `.pk6` to `.pk7` using PKHeX. Edited the Pokemon (that is now in `.pk7` format) to have Hyper Trained stats, and set it to a Box Slot. Decided to dump the `.pk7` of this Pokemon (that keep in mind, originated in Gen 6), and then re-load the `.pk7` into PKHeX.

Upon doing so, PKHeX reverts the changes and its Hyper Trained stats are reverted.

**Comment by kwsch** [comment]
The 'is a file a pk6 vs pk7' logic hasn't been formulated yet. Would be a little complicated as gen < 7 pkm have to be checked for cleared/new values. A pkm can be either:
* Gen6 format
* Gen7 format
* Gen6/7 format (indeterminate -> default to 6)

Currently PKHeX defaults all gen6&prior gameIDs to 6 (the indeterminate case); now that we know what changes are made we can more accurately determine if it is actually in gen7 format (new moves, new properties, cleared properties, etc).

Bank tracks 6 vs 7 in its online storage via a bitflag for a storage slot; unfortunately nothing is stored in the PKM data. Would have been great if they just used a byte to indicate the latest gameID it visited (like Colo/XD), but nooooo.

tl;dr - it's known & todo :)

**Comment by sora10pls** [comment]
Thanks for the clarification!
